cmd/spiffe-helper/main.go

```diff
@@ -13,29 +15,45 @@ func main() {
 	// 1. Create Sidecar
 	// 2. Run Sidecar's Daemon
 
+	log := logger.Std
 	configFile := flag.String("config", "helper.conf", "<configFile> Configuration file path")
 	flag.Parse()
 
 	config, err := ParseConfig(*configFile)
 	if err != nil {
-		log.Fatalf("error parsing configuration file: %v\n%v", *configFile, err)
+		log.Errorf("error parsing configuration file: %v\n%v", *configFile, err)
+		panic(err)
+	}
+	timeout, err := GetTimeout(config)
+	if err != nil {
+		log.Errorf("error parsing timeout: %s\n%v", config.Timeout, err)
+		panic(err)
 	}
+	config.Log = log
 
-	log.Printf("Connecting to agent at %q\n", config.AgentAddress)
+	log.Infof("Connecting to agent at %q\n", config.AgentAddress)
 	if config.Cmd == "" {
-		log.Println("Warning: no cmd defined to execute.")
+		log.Warnf("No cmd defined to execute.")
 	}
 
-	log.Printf("Using configuration file: %q\n", *configFile)
+	log.Infof("Using configuration file: %q\n", *configFile)
 
-	spiffeSidecar, err := sidecar.NewSidecar(config)
+	spiffeSidecar := sidecar.NewSidecar(config)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Why are we applying a timeout to the context used to start the daemon? This will cause the watcher run in the goroutine of RunDaemon to stop watching after five seconds. Also, since we select on this context below, the process will exit.

Reply:
The timeout functionality is existing. I think this was an attempt to move this functionality over from the v1 client. Here is the relevant snippet from the current code base: https://github.com/spiffe/spiffe-helper/blob/3c14013b4adf3b622f95b9367c4842ab0d26b806/pkg/sidecar/sidecar.go#L157-L160

I cant find `X509ClientConfig`, was it moved or deprecated?

Reply:
The workload package referenced there isn't a go-spiffe v1 thing, but rather a different workload api client package that existed in SPIRE. The timeout there would cause a failure to be returned from the Start() method if a successful response had not been received from the workload api in that time. The existing sidecar code just logs errors returned from start and tries again after some backoff. We don't need to replicate that in the new code since we are already logging errors in the watcher implementation and the workload api client internally performs reasonable backoff.